rust/src/dns/dns.rs

```diff
@@ -141,6 +171,12 @@ pub struct DNSQueryEntry {
     pub rrclass: u16,
 }
 
+impl SizeOf for DNSQueryEntry {
+    fn sizeof_inner(&self) -> usize {
+        self.name.len()
```

Comment:
iirc a vector can also tell you how much space it reserved, right?

Reply:
also, would it be possible to somehow handle every Vec<u8> in a generic way, so w/o needing to create small util funcs for each field?

Reply:
> iirc a vector can also tell you how much space it reserved, right?

Yeah, typically they double in size every time they need to grow.  So we could use the `capacity()` method instead.

Reply:
> also, would it be possible to somehow handle every `Vec<u8>` in a generic way, so w/o needing to create small util funcs for each field?

Its actually a util method per struct, not field.  But this is where the derive crate would come into play.  It could look at the data structure and generate the sizeof methods for most of these.